src/modules/chat_commands/index.js

```diff
@@ -234,46 +230,21 @@ function handleCommands(message) {
     return false;
 }
 
-let twitchSendMessage;
-
-function sendMessage() {
-    try {
-        const message = this.get('room.messageToSend');
-        const result = handleCommands(message);
-        if (result === false) {
-            this.set('room.messageToSend', '');
-            return;
-        }
-        if (anonChat.enabled) {
-            twitch.sendChatAdminMessage('You can\'t send messages when Anon Chat is enabled. Type /join or disable Anon Chat in options.');
-            this.set('room.messageToSend', '');
-            return;
-        }
-        if (typeof result === 'string') this.set('room.messageToSend', result);
-    } catch (e) {
-        Raven.captureException(e);
-        debug.log(e);
-    }
-
-    twitchSendMessage.apply(this, arguments);
-}
-
 class ChatCommands {
     constructor() {
-        watcher.on('load.chat_settings', () => this.load());
+        watcher.on('chat.send_message', sendState => this.handleSendMessage(sendState));
     }
 
-    load() {
-        const emberView = twitch.getEmberView($(TEXTAREA_SELECTOR).attr('id'));
-        if (!emberView) return;
-
-        const newTwitchSendMessage = emberView._actions.sendMessage;
-
-        // check if we've already monkeypatched
-        if (newTwitchSendMessage === sendMessage) return;
+    handleSendMessage(sendState) {
+        const result = handleCommands(sendState.message);
+        if (result === false) {
+            sendState.message = '';
```

Comment:
What if you want to abort the message send but not clear the textbox? Or are you proposing a third function that does both?